e2e-tests/src/test/staking.rs

```diff
@@ -1,158 +1,125 @@
+fn nominate(address: &str, nominator_key_pair: &KeyPair, nominee_key_pair: &KeyPair) {
     let nominee_account_id = AccountId::from(nominee_key_pair.public());
-    let connection = create_connection(address).set_signer(nominator_key_pair);
+    let connection = create_connection(address).set_signer(nominator_key_pair.clone());
 
-    let nominate_extrinsic =
-        connection.staking_nominate(vec![GenericAddress::Id(nominee_account_id.clone())]);
-    let finalized_block_hash = connection
-        .send_extrinsic(nominate_extrinsic.hex_encode(), XtStatus::InBlock)
-        .expect("Could not send extrinsic")
-        .expect("Could not get tx hash");
-    info!(
-        "[+] Nominator account {} nominated account {} in finalized {} block.",
-        nominator_account_id, nominee_account_id, finalized_block_hash
-    );
+    let xt = connection.staking_nominate(vec![GenericAddress::Id(nominee_account_id)]);
+    send_xt(&connection, xt.hex_encode(), "nominate");
 }
 
-fn get_top_finalized_block<F>(
-    address: &str,
-    sender_account_key_pair: KeyPair,
-    predicate: F,
-) -> anyhow::Result<Header>
-where
-    F: Fn(&Header) -> bool,
-{
-    let connection = create_connection(address).set_signer(sender_account_key_pair);
-    let (sender, receiver) = channel();
-    connection.subscribe_finalized_heads(sender)?;
-
-    while let Ok(header) = receiver
-        .recv()
-        .map(|h| serde_json::from_str::<Header>(&h).unwrap())
-    {
-        if predicate(&header) {
-            return Ok(header);
-        }
-    }
-    Err(anyhow!("Failed to receive finalized head block"))
+fn payout_stakers(address: &str, validator: KeyPair, era_number: BlockNumber) {
+    let account = AccountId::from(validator.public());
+    let connection = create_connection(address).set_signer(validator);
+    let xt = compose_extrinsic!(connection, "Staking", "payout_stakers", account, era_number);
+
+    send_xt(&connection, xt.hex_encode(), "payout_stakers");
 }
 
-fn wait_for_full_reward_era(
-    address: &str,
-    sender_account_key_pair: KeyPair,
-) -> anyhow::Result<u32> {
-    let current_finalized_block =
-        get_top_finalized_block(address, sender_account_key_pair.clone(), |_| true)?;
-    // TODO: this should be expressed in terms of runtime chain configuration, not hardcoded values
-    // 30 = block_in_session(10) * sessions_in_era (3)
-    const BLOCKS_PER_ERA: u32 = 30;
-
-    let next_full_era_block_number =
-        ((current_finalized_block.number + BLOCKS_PER_ERA) / BLOCKS_PER_ERA + 1) * BLOCKS_PER_ERA;
+fn wait_for_next_era(connection: &Connection) -> anyhow::Result<BlockNumber> {
+    let sessions_per_era: u32 = connection
+        .get_storage_value("Elections", "SessionsPerEra", None)
+        .unwrap()
+        .unwrap();
+    let current_era: u32 = connection
+        .get_storage_value("Staking", "ActiveEra", None)
+        .unwrap()
+        .unwrap();
+    let next_era = current_era + 1;
```

Comment:
I think we need to wait one session more, `let next_era = current_era + 2`. This is because 
* in the `current era`, we set nominators and validators but settings are not yet in place, ie they are waiting to be elected but not yet participating in reward mechanism
* in `current_era + 1`, nominators and validators are working on rewards, and we need full era for that,
* in `current_era + 2`, we can schedule rewards for `current_era + 1`.
Please note how it was in previous version (https://github.com/Cardinal-Cryptography/aleph-node/commit/5fde521d183fdbdf27a9bd120008961b2d4dfb85#diff-ea9a4b230f3d7675b407201681c67f5708ab08d60192844a3c8935d9c4330c39R122) , yet this version us much better as previous one was obfuscated.

Reply:
I agree, I simplified too much ;) fixed